service/src/intTest/java/uk/nhs/adaptors/gp2gp/ehr/SendEhrExtractCoreComponentTest.java

```diff
@@ -0,0 +1,87 @@
+@ExtendWith({SpringExtension.class, MongoDBExtension.class, ActiveMQExtension.class, MockitoExtension.class})
+@SpringBootTest
+@DirtiesContext
+public class SendEhrExtractCoreComponentTest extends BaseTaskTest {
+    @MockBean
+    private WebClient.RequestHeadersSpec<?> request;
+    @MockBean
+    private MhsRequestBuilder mhsRequestBuilder;
+    @MockBean
+    private MhsClient mhsClient;
+    @MockBean
+    private SendEhrExtractCoreTaskDefinition sendEhrExtractCoreTaskDefinition;
+
+    @Autowired
+    private SendEhrExtractCoreTaskExecutor sendEhrExtractCoreTaskExecutor;
+    @Autowired
+    private EhrExtractStatusRepository ehrExtractStatusRepository;
+
+    @Test
+    public void When_NewExtractCoreTask_Expect_DatabaseUpdated() {
+        var ehrExtractStatus = EhrExtractStatusTestUtils.prepareEhrExtractStatus();
+        ehrExtractStatusRepository.save(ehrExtractStatus);
+
+        when(sendEhrExtractCoreTaskDefinition.getConversationId()).thenReturn(ehrExtractStatus.getConversationId());
+        when(sendEhrExtractCoreTaskDefinition.getTaskId()).thenReturn(UUID.randomUUID().toString());
+        when(mhsClient.sendEhrExtractCore(request, sendEhrExtractCoreTaskDefinition)).thenReturn("Successful Mhs Outbound Request");
+
+        sendEhrExtractCoreTaskExecutor.execute(sendEhrExtractCoreTaskDefinition);
+
+        var ehrExtractUpdated = ehrExtractStatusRepository.findByConversationId(ehrExtractStatus.getConversationId()).get();
+        assertThatInitialRecordWasUpdated(ehrExtractUpdated, ehrExtractStatus);
+    }
+
+    @Test
+    public void When_ExtractCoreThrowsException_Expect_EhrExtractStatusNotUpdated() {
+        var ehrExtractStatus = EhrExtractStatusTestUtils.prepareEhrExtractStatus();
+        ehrExtractStatusRepository.save(ehrExtractStatus);
+
+        when(sendEhrExtractCoreTaskDefinition.getConversationId()).thenReturn(ehrExtractStatus.getConversationId());
+        when(sendEhrExtractCoreTaskDefinition.getTaskId()).thenReturn(UUID.randomUUID().toString());
+        doThrow(InvalidOutboundMessageException.class)
+            .when(mhsRequestBuilder).buildSendEhrExtractCoreRequest(sendEhrExtractCoreTaskDefinition);
+
+        assertThrows(InvalidOutboundMessageException.class, () -> sendEhrExtractCoreTaskExecutor.execute(sendEhrExtractCoreTaskDefinition));
+
+        var ehrExtractUpdated = ehrExtractStatusRepository.findByConversationId(ehrExtractStatus.getConversationId()).get();
+        assertThat(ehrExtractUpdated.getEhrExtractCore()).isNull();
+    }
+
+    private void assertThatInitialRecordWasUpdated(EhrExtractStatus ehrExtractStatusUpdated, EhrExtractStatus ehrExtractStatus) {
+        assertThat(ehrExtractStatusUpdated.getUpdatedAt()).isNotEqualTo(ehrExtractStatus.getUpdatedAt());
+        var gpcAccessStructured = ehrExtractStatusUpdated.getEhrExtractCore();
+        assertThat(gpcAccessStructured.getSentAt()).isNotNull();
+        assertThat(gpcAccessStructured.getTaskId()).isNotNull();
```

Comment:
```suggestion
        var ehrExtractCore = ehrExtractStatusUpdated.getEhrExtractCore();
        assertThat(ehrExtractCore.getSentAt()).isNotNull();
        assertThat(ehrExtractCore.getTaskId()).isNotNull();
```

Reply:
@msolecka I have never seen this "Suggested change" feature before!